osf/tasks.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import re
+
+from django.db.models import Model
+
+from framework.celery_tasks import app
+
+logger = logging.getLogger(__name__)
+iri_regex = re.compile(r'http://[^/]+/(?P<id>\w{5})/?')
+
+def get_object_by_url(url: str, model: type[Model]) -> Model | None:
+    if match := iri_regex.match(url):
+        return model.objects.get(guids___id=match['id'])
+    raise Exception(f"Invalid URL received: {url=}")
```

Comment:
This looks good but I would recommend catching specific errors; also see the other comment on using `get_object_by_url()`.

```python
if not match := iri_regex.match(url):
    # log error (maybe sentry too?)
    return None
try:
    return model.objects.get(guids___id=match['id'])
except model.DoesNotExist:
    # log error (maybe sentry too?)
    return None
```